atc/exec/task_step.go

```diff
@@ -277,6 +286,8 @@ func (step *TaskStep) Run(ctx context.Context, state RunState) error {
 
 	go func() {
 		processStatus, processErr = process.Wait()
+		logger.Info("Decreasing active tasks on worker.")
```

Comment:
We would also want to decrement the number of active tasks in [here](https://github.com/concourse/concourse/blob/7d00d54b689eb85a1def93e529f6edbedfabf1f8/atc/exec/task_step.go#L227-L244)

If the task was scheduled on a worker, but then the ATC died and another ATC picked up the task, it would go through this flow. If the task has already completed on the worker, we want to decrement the number of tasks to reflect reality.